### ISSUE TITLE
1.0 segfault in igOpenGL3RenderDrawData

### ISSUE THREAD

**Issue by kraptor** [nimgl/nimgl#25]
Hi, I've tried to no avail to use 1.0, but I get this sigfault:

```
C:\Users\kraptor\.nimble\pkgs\nimgl-#1.0\nimgl\imgui\impl_opengl.nim(243) igOpenGL3RenderDrawData
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

Offending lines are:

```
        if pcmd.userCallback != nil:
            pcmd.userCallback(cmd_list, pcmd.addr)
```

I checked the userCallback, and are populated and pcmd.addr is pointing to ImDrawCmd... so no idea what's going on. If I comment that line, it works but there are lots of mising texts, etc.

**Comment by lmariscal** [comment]
I was having a similar problem, cleaning my nimcache and checking that cimgui versions are uptodate (cimgui.dll and cimgui submodule) fixed it.

**Comment by kraptor** [comment]
Can't fix the issue. I don't know what I'm doing wrong. Using Nim 0.20 + Cimgui 1.71 and Windows. 😢

**Comment by lmariscal** [comment]
It took me a while to replicate this problem, will be looking into it.

**Comment by lmariscal** [comment]
For some reason nimble pulls the submodule versions that are referenced in the `master` branch and doesn't update the submodule versions of the branch. I got a hard time replicating it mostly because I always use the source instead of nimble.
To fix this issue please either clone the repository and verify that `src/nimgl/private/cimgui/cimgui` says at the top that it is from 1.71. Or use [nimgl/imgui](https://github.com/nimgl/imgui) it contains the latest cimgui version on the master branch which would be respected.
I'm going to talk with dom96 the maintainer of nimble in IRC and if it requires open an issue in the nimble repo.

**Comment by kraptor** [comment]
I'll check this out once I get home. Thanks mate!

**Comment by lmariscal** [comment]
@kraptor side note, did you did this with NimGL 😜? 

![](https://user-images.githubusercontent.com/7249728/60570947-e6787f80-9d72-11e9-8b26-d189f44b1256.gif)

And if you did, could I have permission to use it as one of the examples in my BSc Thesis paper?

**Comment by kraptor** [comment]
@lmariscal yes, of course, go ahead. I posted the UI code in a gist not long ago a per a reddit user request:

https://gist.github.com/kraptor/517443a2645b54ccdc7dbf2017a837b3

Feel free to use anywhere you need.

**Comment by kraptor** [comment]
@lmariscal any update on this?

If not, mind to tell me how to install directly the nimgui/cimgui repo? I'm using nimble to manage everything-

Thanks!

**Comment by lmariscal** [comment]
@kraptor you can follow the progress(?) in https://github.com/nim-lang/nimble/issues/675, it sadly falls a little out of my own work.

I would personally recommend to add (nimgl/imgui)[https://github.com/nimgl/imgui] to your project, but if you would like to manage everything with just the core nimgl module follow this steps:

- Download the latest [cimgui](https://github.com/cimgui/cimgui) as a zip or git.
- Find your nimble pkg folder. (Mine: ~/.nimble/pkgs/nimgl-#1.0).
- Go into your nimgl pkg folder and cd into: "nimgl/private/cimgui/".
- Extract or copy and replace the cimgui files you downloaded from cimgui to this folder.
- It should work now.

**Comment by kraptor** [comment]
Seems the bug is fixed upstream... but I wasn't able to test yet.

**Comment by lmariscal** [comment]
Nimble released version [0.11.0](https://github.com/nim-lang/nimble/releases/tag/v0.11.0) which includes a fix to this.
